### DISCUSSION TITLE
How to restart/init-model a training with numb_fparam=1 ?

### DISCUSSION THREAD

**Discussion by Lewis-YL** [deepmodeling/deepmd-kit#1689] [Q&A] [answered]
I can train from the beginning, but I cannot continue the training with --restart or --init-model.

The error produced is:
"No data stat result. one should do data statisitic, before build"

I have "numb_fparam": 1 in input.json during training.
I have fparam.npy in my initial training data.
The checkpoint files I have include: 
model.ckpt-680000.data-00000-of-00001
model.ckpt-680000.index
model.ckpt-680000.meta

**Comment by njzjz** [comment]
This is a bug and I am going to fix it in a few days.

**Comment by njzjz** [comment, marked as answer]
#1695 should fix the issue.

**Reply by Lewis-YL** [reply]
Thank you so much! It works now.

**Comment by phyoung123** [comment]
@Lewis-YL Hi~ I'm preparing to use deepmd-kit to train  a temperature dependent DPMD, but I don't know how to get the ‘fparam.raw’ file before the formal training, would mind give me some references ?  Thank you so much!

**Reply by Lewis-YL** [reply]
When you train the model,

In your input.json,

add
"numb_fparam": 1,
in your 
"fitting net"

**Reply by phyoung123** [reply]
Thank you so much~~, But how should I prepare the 'fparam.raw' file to be accessed by deepmd-kit to start training a TDDP, or it is not necessary to prepare in advance?

**Reply by Lewis-YL** [reply]
you need to prepare it yourself first.

**Reply by lnnbig** [reply]
> Thank you so much~~, But how should I prepare the 'fparam.raw' file to be accessed by deepmd-kit to start training a TDDP, or it is not necessary to prepare in advance?

May I inquire if you have resolved the fparam issue? I am also confused by it. 

Firstly, according to the file found at this link: https://github.com/deepmodeling/deepmd-kit/blob/master/examples/fparam/lmp/in.lammps, the line "pair_style deepmd frozen_model.pb fparam 0.25852028" suggests that the unit for fparam is clearly expressed in electron volts (eV). However, when I generated the Lammps file using DP-GEN, the unit indicated was in Kelvin (K).

Furthermore, there seems to be a significant problem with the value of max_devi_f in the model_devi.out file, which is remarkably small (1e-14). This suggests that there may be bugs present in both DPGEN (Version: 0.9.2) and DeePMD (v2.0.1).

**Reply by J-You** [reply]
> Furthermore, there seems to be a significant problem with the value of max_devi_f in the model_devi.out file, which is remarkably small (1e-14). This suggests that there may be bugs present in both DPGEN (Version: 0.9.2) and DeePMD (v2.0.1).

I can reproduce this error. Could you please tell me how to fix it?

**Reply by lnnbig** [reply]
Hi J-You, have you solved this? I saw your message in the 机器学习势 (MLP) QQ group. hahaha

**Reply by J-You** [reply]
Yes, it is solved. I retrain five models with different seeds.

**Reply by goldenhaaa** [reply]
> 非常感谢~~，但是我应该如何准备 deepmd-kit 访问的 'fparam.raw' 文件才能开始训练 TDDP，还是没有必要提前准备呢？

May I ask how you prepare the **fparam.npy** file before training? I also encountered this problem.
